test/code/scxcorelib/pal/scxfile_test.cpp

```diff
@@ -53,6 +53,8 @@ class SCXFileTest : public CPPUNIT_NS::TestFixture {
     CPPUNIT_TEST( TestSeek );
 #if !defined(DISABLE_WIN_UNSUPPORTED)
     CPPUNIT_TEST( TestCreateTempFile );
+    CPPUNIT_TEST( TestCreateTempFileValidDirectory );
```

Comment:
This is a nice start, but I'd like to see you create a temp file in a non-default directory. To do that, create a new unit test with a directory off of the CWD (current working directory). Create it if needed and delete it in the test when done. This insures that your code works properly without /tmp, but with something else.